Error building lua.cross

### Expected behavior
Build lua.cross without errors.

### Actual behavior
Fails with following errors.

```
app/lua/lauxlib.c: In function 'luaL_assertfail':
app/lua/lauxlib.c:810:3: warning: implicit declaration of function 'dbg_printf' [-Wimplicit-function-declaration]
   dbg_printf("ASSERT@%s(%d): %s\n", file, line, message); 
   ^
.build/cross-lua/app__lua__luac_cross__loslib.o: In function `os_tmpname':
loslib.c:(.text+0x28c): warning: the use of `tmpnam' is dangerous, better use `mkstemp'
.build/cross-lua/app__lua__lauxlib.o: In function `panic':
lauxlib.c:(.text+0xfb): undefined reference to `dbg_printf'
.build/cross-lua/app__lua__lauxlib.o: In function `luaL_assertfail':
lauxlib.c:(.text+0x17a0): undefined reference to `dbg_printf'
collect2: error: ld returned 1 exit status
[COMPILE] .build/cross-lua/app__lua__lapi.o
[COMPILE] .build/cross-lua/app__lua__lauxlib.o
[COMPILE] .build/cross-lua/app__lua__lbaselib.o
[COMPILE] .build/cross-lua/app__lua__lcode.o
[COMPILE] .build/cross-lua/app__lua__ldblib.o
[COMPILE] .build/cross-lua/app__lua__ldebug.o
[COMPILE] .build/cross-lua/app__lua__ldo.o
[COMPILE] .build/cross-lua/app__lua__ldump.o
[COMPILE] .build/cross-lua/app__lua__lfunc.o
[COMPILE] .build/cross-lua/app__lua__lgc.o
[COMPILE] .build/cross-lua/app__lua__llex.o
[COMPILE] .build/cross-lua/app__lua__lmathlib.o
[COMPILE] .build/cross-lua/app__lua__lmem.o
[COMPILE] .build/cross-lua/app__lua__loadlib.o
[COMPILE] .build/cross-lua/app__lua__lobject.o
[COMPILE] .build/cross-lua/app__lua__lopcodes.o
[COMPILE] .build/cross-lua/app__lua__lparser.o
[COMPILE] .build/cross-lua/app__lua__lrotable.o
[COMPILE] .build/cross-lua/app__lua__lstate.o
[COMPILE] .build/cross-lua/app__lua__lstring.o
[COMPILE] .build/cross-lua/app__lua__lstrlib.o
[COMPILE] .build/cross-lua/app__lua__ltable.o
[COMPILE] .build/cross-lua/app__lua__ltablib.o
[COMPILE] .build/cross-lua/app__lua__ltm.o
[COMPILE] .build/cross-lua/app__lua__lundump.o
[COMPILE] .build/cross-lua/app__lua__lvm.o
[COMPILE] .build/cross-lua/app__lua__lzio.o
[COMPILE] .build/cross-lua/app__lua__luac_cross__luac.o
[COMPILE] .build/cross-lua/app__lua__luac_cross__loslib.o
[COMPILE] .build/cross-lua/app__lua__luac_cross__print.o
[COMPILE] .build/cross-lua/app__lua__..__modules__linit.o
[COMPILE] .build/cross-lua/app__lua__..__libc__c_stdlib.o
[LINK] luac.cross
[builder] Error building target
[builder] Last executed command was: 
gcc -o luac.cross .build/cross-lua/app__lua__lapi.o .build/cross-lua/app__lua__lauxlib.o .build/cross-lua/app__lua__lbaselib.o .build/cross-lua/app__lua__lcode.o .build/cross-lua/app__lua__ldblib.o .build/cross-lua/app__lua__ldebug.o .build/cross-lua/app__lua__ldo.o .build/cross-lua/app__lua__ldump.o .build/cross-lua/app__lua__lfunc.o .build/cross-lua/app__lua__lgc.o .build/cross-lua/app__lua__llex.o .build/cross-lua/app__lua__lmathlib.o .build/cross-lua/app__lua__lmem.o .build/cross-lua/app__lua__loadlib.o .build/cross-lua/app__lua__lobject.o .build/cross-lua/app__lua__lopcodes.o .build/cross-lua/app__lua__lparser.o .build/cross-lua/app__lua__lrotable.o .build/cross-lua/app__lua__lstate.o .build/cross-lua/app__lua__lstring.o .build/cross-lua/app__lua__lstrlib.o .build/cross-lua/app__lua__ltable.o .build/cross-lua/app__lua__ltablib.o .build/cross-lua/app__lua__ltm.o .build/cross-lua/app__lua__lundump.o .build/cross-lua/app__lua__lvm.o .build/cross-lua/app__lua__lzio.o .build/cross-lua/app__lua__luac_cross__luac.o .build/cross-lua/app__lua__luac_cross__loslib.o .build/cross-lua/app__lua__luac_cross__print.o .build/cross-lua/app__lua__..__modules__linit.o .build/cross-lua/app__lua__..__libc__c_stdlib.o  -lm
ERROR: Build failed: exit code 1
```


### Test code
Clone current `dev` branch and build lua.cross
```bash
lua tools/cross-lua.lua
```
### NodeMCU version
Commit: 11ded3fc9a70769d28b4be06f6910a54de090c73

### Hardware
No special hardware needed. Same code used to build fine in my build environment with commit 0ee1ab68b009a6413b4f2a61aa6520026c708bf0 now it errors. No changes to my docker image used to build.

Error looks to be caused by commit: ebb537c50273928b2412531fefcec08e1d6ce33d via PR #1564

--- a/tools/cross-lua.lua
+++ b/tools/cross-lua.lua
@@ -16,7 +16,7 @@ end
 builder:init( args )
 builder:set_build_mode( builder.BUILD_DIR_LINEARIZED )
 local output = 'luac.cross'
-local cdefs = '-DLUA_CROSS_COMPILER'
+local cdefs = '-DLUA_CROSS_COMPILER -Ddbg_printf=printf'
 
 -- Lua source files and include path
 local lua_files = [[

